chainerx_cc/chainerx/routines/evaluation.cc

```diff
@@ -0,0 +1,44 @@
+#include "chainerx/routines/evaluation.h"
+
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+
+#include <absl/types/optional.h>
+
+#include "chainerx/array.h"
+#include "chainerx/routines/arithmetic.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/routines/indexing.h"
+#include "chainerx/routines/logic.h"
+#include "chainerx/routines/manipulation.h"
+#include "chainerx/routines/reduction.h"
+#include "chainerx/routines/sorting.h"
+#include "chainerx/routines/statistics.h"
+#include "chainerx/routines/type_util.h"
+#include "chainerx/scalar.h"
+#include "chainerx/shape.h"
+
+namespace chainerx {
+
+Array Accuracy(const Array& x1, const Array& x2, const absl::optional<int8_t>& ignore_label) {
+    if (ignore_label.has_value()) {
+        Array ignore = chainerx::FullLike(x2, Scalar{*ignore_label});
+        Array mask = Equal(x2, ignore);
+        Array ignore_cnt = Sum(mask);
+        Array pred = Where(mask, ignore, ArgMax(x1, 1).Reshape(x2.shape()));
+        Array count = Sum(Equal(pred, x2)) - ignore_cnt;
+        Scalar size{x2.GetTotalSize()};
+        Scalar total = size - AsScalar(ignore_cnt);
+        if (total == 0.0) {
+            return Array{0}.AsType(x1.dtype());
```

Comment:
I'm not sure what constructor you're trying to call this `0` will be interpreted as a `nullptr` and will yield an error. How about using `Zeros`/`ZerosLike` ? Also, could you add tests for this code path?

Reply:
I have changed it to use `Zeros` now. I have added 2 cases where this code path will be taken.
I have added a TODO to make a cleaner interface for this test though.

Reply:
As targets are randomly generated, you are not necessarily testing this code path just by using a fixed ignore_label. You need to construct such a case explicitly where all targets have the ignore_label.

Reply:
@hvy I had also added shapes `(10, 1), (10, )` and `(5, 1), (5, )` for this purpose. So, when `ignore_label` is 0,  `total` will be 0 for the above 2 cases and this code path will be entered. Please let me know if I am still missing something.

Reply:
Oh yes. Sorry.
I have now set `t` to also have all zeros so that for input shapes `(10, 1)` and `(5, 1)`, this code path will be taken.